### PR TITLE
♻️ refactor repo

### LINKED ISSUE #8: Refactor repo structure
**Description**
---

Cleanup repo to setup a standard library structure for better importing.

**Acceptance Criteria**
---

- [x] reorganize directories for library import
- [x] restructure directory OZ style

### PR DESCRIPTION
**Pull request by tekkac**
closes #8

### CODE REVIEW

**Review by bal7hazar — CHANGES_REQUESTED** [review]
In my perspective, presets must be contracts not libraries.
That's why I think the following change must be done:
- rename current `presets`  to `libs`
- rename current `examples`  to `presets`

File: src/metadata/libs/metadata_descriptor_base.cairo [hunk 1]
```diff
@@ -132,7 +134,7 @@ namespace ERC3525MetadataDescriptor {
         bitwise_ptr: BitwiseBuiltin*,
         range_check_ptr,
         instance,
-    }(slot: Uint256) -> (uri: String) {
+    }(slot: Uint256) -> (uri: y) {
```
Comment (bal7hazar) [inline src/metadata/libs/metadata_descriptor_base.cairo]: Typo?

Comment (tekkac) [reply src/metadata/libs/metadata_descriptor_base.cairo]: good catch, fixed

**Review by bal7hazar — APPROVED** [review]
LGTM

**Pull request merged**